test/test_jit.py

```diff
@@ -7892,6 +7892,17 @@ def test_as_tensor_tensor_input(input):
         inp = torch.randn(3, 4)
         self.checkScript(test_as_tensor_tensor_input, (inp,))
 
+    def test_torch_tensor_dtype(self):
+        def foo(s: float):
+            return torch.tensor(s)
```

Comment:
For fun and safety: `return torch.tensor(s), torch.tensor((s,))`?

Reply:
Good point